frontend/device/kindle/device.lua

```diff
@@ -204,10 +204,40 @@ function Kindle:supportsScreensaver()
     end
 end
 
+function Kindle:init()
+    -- Check if the device supports deep sleep/quick boot
+    if lfs.attributes("/sys/devices/platform/falconblk/uevent", "mode") == "file" then
+        -- Now, poke the appreg db to see if it's actually *enabled*...
+        -- NOTE: The setting is only available on registered devices, as such, it *can* be missing,
+        --       which is why we check for it existing and being *disabled*, as that ensures user interaction.
+        local SQ3 = require("lua-ljsqlite3/init")
+        local appreg = SQ3.open("/var/local/appreg.db", "ro")
+        local hibernation_disabled = tonumber(appreg:rowexec(
+            "SELECT EXISTS(SELECT value FROM properties WHERE handlerId = 'dcc' AND name = 'hibernate.enabled' AND value = 0);"
+        ))
+        -- Check the actual delay while we're there...
+        local hibernation_delay = appreg:rowexec(
+            "SELECT value FROM properties WHERE handlerId = 'dcc' AND name = 'hibernate.s2h.rtc.secs'") or
+            appreg:rowexec("SELECT value FROM properties WHERE handlerId = 'dcd' AND name = 'hibernate.s2h.rtc.secs'") or
+            3600
+        appreg:close()
+        if hibernation_disabled == 1 then
+            self.canDeepSleep = false
+        else
+            self.canDeepSleep = true
+            self.hibernationDelay = tonumber(hibernation_delay)
+            logger.dbg("Kindle: Device supports hibernation, enters hibernation after", self.hibernationDelay, "seconds in suspend")
+        end
```

Comment:
Looks a bit brittle (in the what if they change it sense). What happens if the QUERY says that table doesn't exist or something?

Reply:
For the first query, EXISTS returns 0 and we assume ti's enabled, and for the other two, we fall back to 1H ;).

Reply:
(Not a SQL wizard, so do feel free to poke hole in my assumptions ;p)

Reply:
I mean what if the query fueling hibernation_delay is effectively invalid, does it just return an otherwise harmless 1? It's not the SQL I'm asking about but how SQLite interacts with Lua here. :-)

Reply:
Oh, `rowexec` returns `nil` on query failure, hence the chain of `or`s ;).

(And it throws an `error` on a malformed query).

Reply:
Alright, nil is good enough.